### PR TITLE
Added liking endpoint

### PR DESCRIPTION
**Pull request by scholvat**
Adding an end point for liking a post. Currently this is done using updatePost endpoint, which is not very secure because then any user can edit any part of the post. There will be a corresponding change to the client sides, but it will be dependent on this PR: https://github.com/CGUC/skybunk-mobile/pull/143. This PR will be merged before PR #72, since that PR needs these updates to function properly.

### CODE REVIEW

File: models/Posts.js [hunk 1]
```diff
@@ -199,6 +199,41 @@ PostSchema.statics.updatePost = function (id, postData) {
...
+
+  return new Promise((resolve, reject) => {
+    this.findById(id).then(post => {
+      if (post && post.usersLiked) {
+        if(isLiked){
+          if(post.usersLiked.some(e => e == user)){
+            reject("Already liked");
+          }else{
+            post.usersLiked.push(user);
+            post.likes = post.usersLiked.length;
```
Comment (neilbrub) [inline models/Posts.js]: :ok_hand:

File: controllers/postController.js [hunk 2]
```diff
@@ -86,6 +86,20 @@ router.put('/:id', verifyToken, (req, res) => {
   });
 });
 
+/**
+ * Like a post
+ */
+router.post('/:pid/like', verifyToken, (req, res) => {
+  Post.likePost(req.params.pid, req.user._id, req.body.isLiked)
```
Comment (neilbrub) [inline controllers/postController.js]: Any way we could rename `isLiked` to `addLike`? To me `isLiked` suggests we're flagging whether this user has already liked the post (how we interpret it client-side at the moment), when in fact what we're really asking is "should we add or take away a like?". Lmk if I've misunderstood the code though :stuck_out_tongue_closed_eyes:

**Comment by scholvat** [comment]
@neilbrub Good point about isLike, it is now changed to addLike.

**Comment by scholvat** [comment]
On a bit of a tangent, is there a reason the client uses post.likes and usersLiked? Why doesn't it just use usersLiked.length?

**Comment by neilbrub** [comment]
> On a bit of a tangent, is there a reason the client uses post.likes and post.usersLiked? Why doesn't it just use post.usersLiked.length?

Haha no not really, other than the fact that we started with just the concept of how many likes a post had and only later introduced the concept of tracking who'd liked what (and I guess then didn't acknowledge the obviously easier way of doing it :P). Happy to get rid of the count field altogether :+1:
(isn't it great to have fresh eyes xD)

**Pull request merged**